### DISCUSSION TITLE
Problem with require.context when using v1.5

### DISCUSSION THREAD

**Discussion by Sparclex** [tailwindlabs/headlessui#1203] [Help]
Hi

When using headless ui v1.5 with react `require.context(...)` seems to return a promise instead of a sync response. The problem seems to only exists when using v1.5. Does somebody know a solution to this. I would like to be able to use webpack `require.context` synchronously even with headless ui, since it will be during the nextjs ssg anyways. I created an example project with and without headless ui to present the error. Using yarn for both:

[main branch, nextjs without headlessui](https://github.com/Sparclex/headlessui-require-context-react-test)
[with-headless-ui branch](https://github.com/Sparclex/headlessui-require-context-react-test/tree/with-headless-ui)

Would be great if someone could help me with this or provide a workaround. My goal is to dynamically load components from a directory synchronously. 

Thanks,

Silvan

**Comment by gfortaine** [comment]
@Sparclex It seems that we faced exactly the same concern (`require.context(...)` returning a promise instead of a sync response) albeit with another package this time (axios-retry) : https://github.com/softonic/axios-retry/issues/206
Everything went fine until our upgrade to Next.js 12 and it looks like that we've finally found the guilty option in the configuration :
https://nextjs.org/blog/next-11-1#es-modules-support
Here is a temporary fix :

```js
// next.config.js
module.exports = {
  // Prefer loading of CommonJS over ES Modules
  experimental: { esmExternals: false },
};
